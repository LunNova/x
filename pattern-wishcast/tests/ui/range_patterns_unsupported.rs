// SPDX-FileCopyrightText: 2025 LunNova
//
// SPDX-License-Identifier: MIT

//! Test that range patterns produce helpful error messages

use pattern_wishcast::pattern_wishcast;

pattern_wishcast! {
	enum Value is <P: PatternFields> = {
		Number { value: i32 },
		Flag,
	};

	// Range patterns should produce helpful error messages:
	type SmallNumbers = Value is Number { value: 1..10 };
}

fn main() {}
