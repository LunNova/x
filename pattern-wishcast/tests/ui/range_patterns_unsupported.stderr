error: Field patterns are not supported. Only wildcard pattern { .. } is allowed for struct variants. Field patterns will require native pattern types support in Rust.
  --> tests/ui/range_patterns_unsupported.rs:16:40
   |
16 |     type SmallNumbers = Value is Number { value: 1..10 };
   |                                           ^^^^^
